shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/converters/ManagedBridgeConverter.java

```diff
@@ -6,44 +6,53 @@
 import com.redhat.service.smartevents.infra.core.api.dto.KafkaConnectionDTO;
 import com.redhat.service.smartevents.infra.v2.api.exceptions.definitions.platform.InvalidURLException;
 import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.DNSConfigurationDTO;
 import com.redhat.service.smartevents.shard.operator.v2.resources.DNSConfigurationSpec;
 import com.redhat.service.smartevents.shard.operator.v2.resources.KNativeBrokerConfigurationSpec;
 import com.redhat.service.smartevents.shard.operator.v2.resources.KafkaConfigurationSpec;
 import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
-import com.redhat.service.smartevents.shard.operator.v2.resources.TLSSpec;
+import com.redhat.service.smartevents.shard.operator.v2.resources.SourceConfigurationSpec;
 
 public class ManagedBridgeConverter {
 
     public static ManagedBridge fromBridgeDTOToManageBridge(BridgeDTO bridgeDTO, String namespace) {
-        try {
+        DNSConfigurationSpec dns = fromDNSConfigurationDTOToDNSConfigurationSpec(bridgeDTO.getDnsConfiguration());
 
-            DNSConfigurationSpec dns = DNSConfigurationSpec.Builder.builder()
-                    .host(new URL(bridgeDTO.getEndpoint()).getHost())
-                    .tls(new TLSSpec(bridgeDTO.getTlsKey(), bridgeDTO.getTlsCertificate()))
-                    .build();
+        KafkaConnectionDTO knativeBrokerKafkaConnectionDTO = bridgeDTO.getKnativeBrokerConfiguration().getKafkaConnection();
+        KafkaConfigurationSpec knativeBrokerKafkaConfigurationSpec = fromKafkaConnectionDTOToKafkaConfigurationSpec(knativeBrokerKafkaConnectionDTO);
+        KNativeBrokerConfigurationSpec kNativeBrokerConfigurationSpec = new KNativeBrokerConfigurationSpec(knativeBrokerKafkaConfigurationSpec);
 
-            KafkaConnectionDTO kafkaConnectionDTO = bridgeDTO.getKafkaConnection();
-            KafkaConfigurationSpec kafkaConfigurationSpec = KafkaConfigurationSpec.Builder.builder()
-                    .bootstrapServers(kafkaConnectionDTO.getBootstrapServers())
-                    .password(kafkaConnectionDTO.getClientSecret())
-                    .user(kafkaConnectionDTO.getClientId())
-                    .saslMechanism(kafkaConnectionDTO.getSaslMechanism())
-                    .securityProtocol(kafkaConnectionDTO.getSecurityProtocol())
-                    .topic(kafkaConnectionDTO.getTopic())
-                    .build();
+        KafkaConnectionDTO sourceKafkaConnectionDTO = bridgeDTO.getSourceConfiguration().getKafkaConnection();
+        KafkaConfigurationSpec sourceKafkaConfigurationSpec = fromKafkaConnectionDTOToKafkaConfigurationSpec(sourceKafkaConnectionDTO);
+        SourceConfigurationSpec sourceConfigurationSpec = new SourceConfigurationSpec(sourceKafkaConfigurationSpec);
```

Comment:
I think you are going to NPE as the `sourceConfiguration` is not set in the `BridgeServiceImpl:toDTO`

Reply:
I added check to prevent NPE